### DISCUSSION TITLE
Appending to referenced enum

### DISCUSSION THREAD

**Discussion by martinwangen** [OAI/OpenAPI-Specification#2943] [Q&A] [answered]
Would love to be able to append to an existing enum. Not all values are applickable in all cases, this means that i need to redefine the enum.  Especially when used together with deepmap's oapi-gen this would be nice.

**Example:**
SensorType.yaml:
```yaml
type: string
enum: [Motion, Push, Voice]
```
Doorbell.yaml:
```yaml
    schema:
    $ref: "#/shared/components/schemas/SensorType.yaml"
    description: Sensor to trigger doorbell
```
Alarm.yaml:
```yaml
  schema:
    $ref: "#/shared/components/schemas/SensorType.yaml"
    append: [ Temp, Vibration, SignalLoss ]
    description: Sensor to trigger alarm
```

**Comment by webron** [comment, marked as answer]
This is more of a JSON Schema request than an OpenAPI one. You can achieve what you're looking for using `oneOf` or `anyOf` depending on the content of the different enums.

```yaml
  schema:
    oneOf:
      - $ref: "#/shared/components/schemas/SensorType.yaml"
      - enum: [ Temp, Vibration, SignalLoss ]
    description: Sensor to trigger alarm
```

**Comment by handrews** [comment]
Answered and accepted, closing.
